src/structures/Interaction.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const Base = require('./Base');
+const { InteractionTypes } = require('../util/Constants');
+const SnowflakeUtil = require('../util/SnowflakeUtil');
+
+/**
+ * Represents an interaction.
+ * @extends {Base}
+ */
+class Interaction extends Base {
```

Comment:
There is also the application_id and the version fields in the interaction payload

Reply:
`application_id` isn't needed for gateway interactions because `this.client.application.id` should always be present

`version` is documented as "[always `1`](https://discord.com/developers/docs/interactions/slash-commands#interaction)", so I don't see why it should be added either

Reply:
The version field should be implemented, the library purpose is to support all api features right? In the future it might be usefull.

Reply:
If it ever becomes useful, it can be added to the interaction class later